scripts/repair_batlas_references.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+import argparse
+import itertools
+import json
+import sys
+import transaction
+
+from pleiades.dump import getSite, spoofRequest
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(
+        description="Repair old-style and idiosyncratic BAtlas references on Place objects"
+    )
+    parser.add_argument("input_file", type=str, help="The JSON input file name")
+    parser.add_argument("--dry-run", action="store_true", help="Enable dry run mode")
+    # Deal with run script wrapping. We're passed a -c flag we don't use:
+    parser.add_argument("-c", type=str, help=argparse.SUPPRESS)
+
+    parsed = parser.parse_args()
+
+    return parsed.input_file, parsed.dry_run
+
+
+def ingest_json_file(file_path):
+    try:
+        with open(file_path, "r") as file:
+            data = json.load(file)
+            return data
+    except IOError:
+        print("Error: File {} does not exist or could not be read!".format(file_path))
+        raise
+    except ValueError:
+        print("Error: Invalid JSON data!")
+        raise
+
+
+def chunked_iterable(iterable, size):
+    """Loop over an iterable and return chunks of @size"""
+    it = iter(iterable)
+    while True:
+        chunk = tuple(itertools.islice(it, size))
+        if not chunk:
+            break
+        yield chunk
+
+
+def translate_to_citation(json_record):
+    json_key_to_citation_key = {
+        "accessURI": "access_uri",
+        "alternateURI": "alternate_uri",
+        "bibliographicURI": "bibliographic_uri",
+        "citationDetail": "citation_detail",
+        "formattedCitation": "formatted_citation",
+        "otherIdentifier": "identifier",
+        "shortTitle": "short_title",
+        "type": "type",
+    }
+
+    as_citation = {}
+    for key, value in json_record.iteritems():
```

Comment:
Maybe just use `items()` in case we ever want to use this in Python 3? Then again, the script will probably need other updates like the encoding stuff below (`safe_nativestring`?).

Reply:
That was a VS Code autocomplete aggression. I'll have a look at the p3/p2 question.

Reply:
Punting on wider p3/p2 compat issues, unfortunately.